### PR TITLE
DEV-1339: add rails example

### PR DESCRIPTION
**Pull request by stephencattaneo**
review level: normal
closes DEV-1339

### CODE REVIEW

File: fastapi/README.md [hunk 1]
```diff
@@ -26,6 +29,6 @@ Chrome does not allow access to http urls and will automatically redirect to htt
 1. Open Chrome and type chrome://net-internals/#hsts in the address bar.
 2. In the "Query HSTS/PKP domain" section, enter localhost in the "Domain" field and click "Query".
 3. You should see an entry for localhost with many varaibles set such as `static_upgrade_mode: FORCE_HTTPS`
-4. Enter `localhost` under the Delete domain security policies and click Delete. 
+4. Enter `localhost` under the Delete domain security policies and click Delete.
 5. Refresh your page, making sure you are usring [http](http://localhost:3000)
```
Comment (drbrady8800) [inline fastapi/README.md]:
```suggestion
5. Refresh your page, making sure you are using [http](http://localhost:3000)
```

File: fastapi/README.md [hunk 2]
```diff
@@ -26,6 +29,6 @@ Chrome does not allow access to http urls and will automatically redirect to htt
 1. Open Chrome and type chrome://net-internals/#hsts in the address bar.
 2. In the "Query HSTS/PKP domain" section, enter localhost in the "Domain" field and click "Query".
 3. You should see an entry for localhost with many varaibles set such as `static_upgrade_mode: FORCE_HTTPS`
-4. Enter `localhost` under the Delete domain security policies and click Delete. 
+4. Enter `localhost` under the Delete domain security policies and click Delete.
```
Comment (drbrady8800) [inline fastapi/README.md]: Was the point of adding the mkcert to get rid of the need to do these steps?

File: rails/.gitignore [hunk 3]
```diff
@@ -0,0 +1,6 @@
+# local env files
+.env*.local
+
+# vercel
+.vercel
+certificates/*.pem
```
Comment (drbrady8800) [inline rails/.gitignore]: [nit] EOF newline

File: rails/compose.yml [hunk 4]
```diff
@@ -1,20 +1,19 @@
...
     build: ./backend
     env_file:
       - .env.local
     ports:
       - "8080:8080"
     volumes:
-      - ./backend:/code
-  frontend:
-    container_name: nextjs
+      # - ./backend:/code
```
Comment (drbrady8800) [inline rails/compose.yml:9]: Should this be commented out?

File: rails/compose.yml [hunk 5]
```diff
@@ -1,20 +1,19 @@
...
+      - ./certificates:/code/certificates
+  nextjs:
     build: ./frontend
     env_file:
       - .env.local
     volumes:
-      - ./frontend:/code
+      # - ./frontend:/code
```
Comment (drbrady8800) [inline rails/compose.yml:16]: Should this be commented out?

File: rails/frontend/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+WORKDIR /code
+
+COPY . .
+RUN npm install
+
+EXPOSE 3000
+
+ENV PORT 3000
+
+CMD ["npm", "run", "dev"]
```
Comment (drbrady8800) [inline rails/frontend/Dockerfile]: [nit] EOF newline

**Review by drbrady8800 — APPROVED** [review]
Looks amazing! I pulled it down and ran the example locally, super easy to do! Have a couple nits / questions but nothing blocking.

File: rails/compose.yml (hunk 5 above)
Comment (stephencattaneo) [reply rails/compose.yml:16]: ya thats on purpose. By default we want the containers to run w/o changes from your local env.

File: rails/compose.yml (hunk 4 above)
Comment (stephencattaneo) [reply rails/compose.yml:9]: yes

**Pull request merged**